### ISSUE TITLE
Feature idea: version check on application startup

### ISSUE THREAD

**Issue by SamLoy** [IceBreakerG/ngx-version-check#14]
I have a requirement to do an initial version check when the application starts up, then later perform future version checks on an interval. 

Sometimes I find the browser renders cached versions of the site's javascript, causing the base installed version to remain as an old one.
Setting the `frequency` variable to 5 seconds certainly solves the problem. However then the site is polling far too frequently for "genuine" version updates.

For detecting new versions while the site is active, I would like poll for version changes every 10 minutes. However if the browser has launched an old version, this wouldn't be detected for 10 minutes. So this fork introduces an initial version check on startup.

I have forked the repo and used patch-package to include it into my project for the time being. 

If this feature interests you I am happy to put together a PR + update the tests? Perhaps with an optional flag for activating the startup version check. 

Here is the relevant code change to `projects/version-check/src/lib/version-check.service.ts`: 
``` 
/**
   * Starts the version check interval for the specified frequency.
   * @param config The configuration parameters for the notification function and version check frequency.
   */
  public startVersionChecking(config: IVersionCheck = { notification: null, frequency: 1800000 }) {
    this.checkVersion(config.notification).then(() => {
      if (!this.newVersionAvailable) {
        this.versionCheckInterval = interval(config.frequency).subscribe(() => {
          this.checkVersion(config.notification)
        })
      }
    });
  }

  /** Will do the call and check if the hash has changed or not. */
  private checkVersion(notification: any): Promise<void> {
    // Timestamp these requests to invalidate caches
    return this.http.get(`version.json?t=${new Date().getTime()}`).toPromise().then((response: any) => {
      this.newVersionAvailable = this.hasHashChanged(this.currentHash, response.hash)

      // Stop checking for a new version if a new version is already available
      if (this.newVersionAvailable) {
        this.stopVersionChecking()

        // Call the consuming client's notification method if one exists
        if (notification) notification()
      }
    }).catch(err => {
      console.error(err, 'Error checking version')
    })
  }
```

Check out the fork here: 
https://github.com/SamLoy/ngx-version-check

**Comment by IceBreakerG** [comment]
@SamLoy Hey, if I make the checkVersion() public, you can just do a manual check on start up as well instead of polling it. Would that work for you too?

**Comment by SamLoy** [comment]
Smashing yeah, that would also do the job! 

Much simpler solution 👍

Thanks

**Comment by IceBreakerG** [comment]
Alright. I'll work on that this weekend if I don't forget. Hard to remember what day it is sometimes lol.

**Comment by IceBreakerG** [comment]
@SamLoy I just updated the package to version 1.0.3. The checkVersion() method is public now. To use it, you just need to pass in whichever method you're using in your application to notify the user that there's a new version.

**Comment by SamLoy** [comment]
Great, thanks for the speedy update

**Comment by IceBreakerG** [comment]
@SamLoy There was an issue with the publishing of the last package, version 1.0.3. I didn't notice it until now, when I was working on another project. I just published version 1.0.4 which fixes the issue. If you installed 1.0.3, update to 1.0.4. The issue is that 2 of the folders were missing in the package.
